cpp/src/arrow/flight/sql/server.h

```diff
@@ -85,6 +85,32 @@ struct ARROW_FLIGHT_SQL_EXPORT PreparedStatementUpdate {
   std::string prepared_statement_handle;
 };
 
+/// \brief A bulk ingestion request
+struct ARROW_FLIGHT_SQL_EXPORT StatementIngest {
+  enum IngestMode {
+    kUnspecified,
+    kCreate,
+    kAppend,
+    kReplace,
+    kCreateAppend,
+  };
+
+  /// \brief The ingestion behavior.
+  IngestMode mode;
+  /// \brief The destination table to load into.
+  std::string table;
+  /// \brief The DB schema of the destination table.
+  std::string schema;
```

Comment:
`std::optional<std::string>` for optionals

Reply:
Protobuf should expose `has_schema` etc. so you can check if it was actually absent vs empty string

Reply:
Good point. Looks better now.